contracts/UniV3Vault.sol

```diff
@@ -96,31 +94,54 @@ contract UniV3Vault is IERC721Receiver, Vault {
         emit CollectedEarnings(tx.origin, to, collectedEarnings0, collectedEarnings1);
     }
 
-    /// @inheritdoc Vault
-    function tvl() public view override returns (uint256[] memory tokenAmounts) {
-        tokenAmounts = new uint256[](2);
-        if (uniV3Nft == 0)
-            return tokenAmounts;
+    /// @inheritdoc IVault
+    function tvl() public view returns (uint256[] memory minTokenAmounts, uint256[] memory maxTokenAmounts) {
+        if (uniV3Nft == 0) {
+            return (new uint256[](2), new uint256[](2));
+        }
+        IUniV3VaultGovernance.DelayedProtocolParams memory params = IUniV3VaultGovernance(address(_vaultGovernance))
+            .delayedProtocolParams();
         (
-            , , , , , 
-            int24 tickLower, 
-            int24 tickUpper, 
+            ,
+            ,
+            ,
+            ,
+            ,
+            int24 tickLower,
+            int24 tickUpper,
             uint128 liquidity,
-            , ,
+            ,
+            ,
             uint128 tokensOwed0,
             uint128 tokensOwed1
-        ) = _positionManager().positions(uniV3Nft);
-        (uint160 sqrtPriceX96, , , , , , ) = pool.slot0();
+        ) = params.positionManager.positions(uniV3Nft);
+        (, uint256 minPriceX96, uint256 maxPriceX96) = params.oracle.spotPrice(_vaultTokens[0], _vaultTokens[1]);
+        uint256 minSqrtPriceX96 = CommonLibrary.sqrtX96(minPriceX96);
+        uint256 maxSqrtPriceX96 = CommonLibrary.sqrtX96(maxPriceX96);
         uint160 sqrtPriceAX96 = TickMath.getSqrtRatioAtTick(tickLower);
         uint160 sqrtPriceBX96 = TickMath.getSqrtRatioAtTick(tickUpper);
-        (uint256 amount0, uint256 amount1) = LiquidityAmounts.getAmountsForLiquidity(
-            sqrtPriceX96,
+        (uint256 amountMin0, uint256 amountMin1) = LiquidityAmounts.getAmountsForLiquidity(
+            uint160(minSqrtPriceX96),
+            sqrtPriceAX96,
+            sqrtPriceBX96,
+            liquidity
+        );
+        (uint256 amountMax0, uint256 amountMax1) = LiquidityAmounts.getAmountsForLiquidity(
+            uint160(maxSqrtPriceX96),
             sqrtPriceAX96,
             sqrtPriceBX96,
             liquidity
         );
-        tokenAmounts[0] = amount0 + uint256(tokensOwed0);
-        tokenAmounts[1] = amount1 + uint256(tokensOwed1);
+        minTokenAmounts = new uint256[](2);
+        maxTokenAmounts = new uint256[](2);
+        minTokenAmounts[0] = amountMin0 < amountMax0 ? amountMin0 : amountMax0;
+        minTokenAmounts[1] = amountMin1 < amountMax1 ? amountMin1 : amountMax1;
+        maxTokenAmounts[0] = amountMin0 < amountMax0 ? amountMax0 : amountMin0;
+        maxTokenAmounts[1] = amountMin1 < amountMax1 ? amountMax1 : amountMin1;
```

Comment:
found a few similar cases
https://github.com/mellow-finance/mellow-vaults/pull/44/files#diff-abd93f623f4d42dfd0b0a21591138a26a1b2dc5f44b4119f0c801ab64471979cR225
let's add following function to the CommonLibrary
```solidity
function minUint256(uint256 a, uint256 b) internal pure returns (uint256) {
    return a < b ? a : b;
}
```
:)